example/echo-op/echo_op.cpp

```diff
@@ -14,276 +14,324 @@
+                                  buffer_.capacity() - buffer_.size()),
+                              std::min<std::size_t>(65536,              // and over 65536 is too much.
+                                  buffer_.max_size() - buffer_.size()));
+
+                        // Read some data into our dynamic buffer_. We transfer
+                        // ownership of the composed operation by using the
+                        // `std::move(*this)` idiom. The `yield` keyword causes
+                        // the function to return immediately after the initiating
+                        // function returns.
+
+                        yield stream_.async_read_some(
+                            buffer_.prepare(bytes_to_read), std::move(*this));
+
+                        // After the `async_read_some` completes, control is
+                        // transferred to this line by the `reenter` keyword.
+
+                        // Move the bytes read from the writable area to the
+                        // readable area.
+
+                        buffer_.commit(bytes_transferred);
+
+                        // If an error occurs, deliver it to the caller's completion handler.
+                        if(ec)
+                            break;
+
+                        // Keep looping until we get the newline
+                        continue;
+                    }
+
+                    // We have our newline, so send the first `pos` bytes of the
+                    // buffers. The function `buffers_prefix` returns the front part
+                    // of the buffers we want.
+
+                    yield net::async_write(stream_,
+                        beast::buffers_prefix(pos, buffer_.data()), std::move(*this));
+
+                    // After the `async_write` completes, our completion handler will
+                    // be invoked with the error and the number of bytes transferred,
+                    // and the `reenter` statement above will cause control to jump
+                    // to the following line. The variable `pos` is no longer valid
+                    // (remember that we returned from the function using `yield` above)
+                    // but we can use `bytes_transferred` to know how much of the buffer
+                    // to consume. With "real" coroutines this will be easier and more
+                    // natural.
+
+                    buffer_.consume(bytes_transferred);
+
+                    // The loop terminates here, and we will either deliver a
+                    // successful result or an error to the caller's completion handler.
+
+                    break;
+                }
+
+                // When a composed operation completes immediately, it must not
+                // directly invoke the completion handler otherwise it could
+                // lead to unfairness, starvation, or stack overflow. Therefore,
```

Comment:
I'd just scare people that it's UB. If we're dealing with coroutines, calling the handler directly would resume a coroutine that is running.

Reply:
I copied Chris' wording from the executors paper